transforms/dialog-full-screen-app-wrapper/dialog-full-screen-app-wrapper.js

```diff
@@ -0,0 +1,184 @@
+/*
+ * Add <AppWrapper> around the children on <DialogFullScreen>
+ * <DialogFullScreen><AppWrapper>{ children }</ AppWrapper></ DialogFullScreen>
+ */
+import registerMethods from "../registerMethods";
+
+function transformer(fileInfo, api, options) {
+  const j = api.jscodeshift;
+  registerMethods(j);
+
+  // adds the AppWrapper import path
+  const addImport = () => {
+
+    // DialogFullScreen import
+    const dialogFullScreenImport = root.find(j.ImportDeclaration, {
+      source: {
+        type: "Literal",
+        value: "carbon-react/lib/components/dialog-full-screen"
+      }
+    });
+
+    const appWrapperImportString = "carbon-react/lib/components/app-wrapper";
+
+    // AppWrapper import
+    let appWrapperImport = root.find(j.ImportDeclaration, {
+      source: {
+        type: "Literal",
+        value: appWrapperImportString,
+      },
+    });
+
+    // if no AppWrapper import exists already add one
+    if (!appWrapperImport.size()) {
+      appWrapperImport = j.importDeclaration(
+        [j.importDefaultSpecifier(j.identifier("AppWrapper"))],
+        j.stringLiteral(appWrapperImportString)
+      );
+    
+      // insert AppWrapper import after DialogFullScreen import
+      dialogFullScreenImport.insertAfter(appWrapperImport);
+
+      return true;
+    }
+  };
+
+  // filter the variable declarations against the expression declared as a child of DialogFullScreen
+  const filteredDeclarations = (variableDeclarations, expressionContainerName) => {
+    if (!variableDeclarations.length) {
+      return variableDeclarations;
+    }
+
+    const filteredDeclarations = [];
+
+    variableDeclarations.filter((vd) => {
+      const { declarations } = vd.value;
+
+      if (declarations.length && declarations[0].id.name === expressionContainerName) {
```

Comment:
There's potential for an error here if `declarations[0].id` doesn't exist, I'd probably check for that as well: 
```suggestion
      if (
        declarations.length && 
        declarations[0].id &&
        declarations[0].id.name === expressionContainerName
      ) {
```

Reply:
If there was no id in the AST node it wouldn't be valid JavaScript so I don't think this is necessary.

Reply:
Are you sure? I've seen app breaking errors come out of these situations in the past

Reply:
In this scenario there will have to be an id for the variable/ function declaration

Reply:
Yeah having looked at AST explorer, you're right - no need for this check!